lib/pbench/server/cache_manager.py

```diff
@@ -882,6 +882,14 @@ def create(self, controller_name: str, tarfile: Path) -> Tarball:
         Returns
             Tarball object
         """
+        try:
+            metadata = Tarball._get_metadata(tarfile)
+            controller_name = metadata["run"]["controller"]
+        except Exception as exc:
```

Comment:
I think you have to catch at least MetadataError, and what ever happens when a `metadata.log` file is not found.